### PR TITLE
chain/owasm: Add end-to-end testing code with silly wasm

### PR DESCRIPTION
**Pull request by sorawit**

### CODE REVIEW

File: chain/owasm/res/silly.rs [hunk 1]
```diff
@@ -0,0 +1,44 @@
+extern "C" {
```
Comment (sorawit) [inline chain/owasm/res/silly.rs:1]: This is for you to read the raw rust code. Not actually used.

File: chain/owasm/execute_test.go [hunk 2]
```diff
@@ -13,7 +13,7 @@ func TestExecuteCanCallEnv(t *testing.T) {
 	if err != nil {
 		panic(err)
 	}
-	result, gasUsed, err := Execute(NewMockExecutionEnvironment(), code, "execute", []byte{}, 10000)
+	result, gasUsed, err := Execute(&mockExecutionEnvironment{}, code, "execute", []byte{}, 10000)
 	if err != nil {
```
Comment (taobun) [inline chain/owasm/execute_test.go]: Use `require.NotNil(t, err)` instead if and panic

File: chain/owasm/execute_test.go [hunk 3]
```diff
@@ -26,6 +26,30 @@ func TestExecuteOutOfGas(t *testing.T) {
...
 		panic(err)
 	}
-	_, _, err = Execute(NewMockExecutionEnvironment(), code, "execute", []byte{}, 10)
+	_, _, err = Execute(&mockExecutionEnvironment{}, code, "execute", []byte{}, 10)
 	require.EqualError(t, err, "gas limit exceeded")
 }
+
+func TestExecuteEndToEnd(t *testing.T) {
+	code, err := ioutil.ReadFile("./res/silly.wasm")
+	if err != nil {
```
Comment (taobun) [inline chain/owasm/execute_test.go]: ditto

**Review by taobun — CHANGES_REQUESTED** [review]

File: chain/owasm/execute_test.go (hunk 2 above)
Comment (sorawit) [reply chain/owasm/execute_test.go]: Thanks! Should be `require.Nil` though.

File: chain/owasm/execute_test.go [hunk 4]
```diff
@@ -10,10 +10,8 @@ import (
 
 func TestExecuteCanCallEnv(t *testing.T) {
 	code, err := ioutil.ReadFile("./res/main.wasm")
-	if err != nil {
-		panic(err)
-	}
-	result, gasUsed, err := Execute(NewMockExecutionEnvironment(), code, "execute", []byte{}, 10000)
+	require.Nil(t, err)
+	result, gasUsed, err := Execute(&mockExecutionEnvironment{}, code, "execute", []byte{}, 10000)
 	if err != nil {
```
Comment (taobun) [inline chain/owasm/execute_test.go]: This code too

**Review by taobun — CHANGES_REQUESTED** [review]

File: chain/owasm/execute_test.go (hunk 4 above)
Comment (sorawit) [reply chain/owasm/execute_test.go]: Right. Thanks!

**Review by taobun — APPROVED** [review]
Nice!

**Review by prin-r — APPROVED** [review]

**Pull request merged**